Towards eliminating signed overflows.

jhensel opened the pull request:
At least for some benchmarks of the termination category, overflows are now avoided.

dbeyer commented:
Wouldn't it be great to keep the wrong ones as examples for invalid overflow?

jhensel commented:
Yes, we could do that. But then we remove them from the termination category, don't we?

peterschrammel commented:
@jhensel, have a look how I did it in https://github.com/sosy-lab/sv-benchmarks/pull/357

jhensel commented:
Thanks.
I fixed MinusBuiltIn_true-termination_true-no-overflow.c, and renamed/copied MinusMin_true-termination_true-no-overflow.c into todo.
I do not (yet) see how an overflow of tmp is possible in b.05_true-termination_true-no-overflow.c.

peterschrammel commented:
Sorry, I was imprecise: `2*tmp`  can overflow...

jhensel commented:
But tmp is limited by 1073741823 (smaller than 1073741824) before 2*tmp is computed.

peterschrammel commented:
underflow...

jhensel commented:
Sorry, fixed.

peterschrammel commented:
@jhensel, can you please rebase this PR?

kfriedberger commented on c/termination-restricted-15-todo/MinusMin_true-termination.c:
```diff
@@ -11,13 +11,14 @@ int main() {
         y = y+1;
-        if (y < x) { min = y; }
-        else { min = x; }
+        res = res+1;
```
Why is `res` needed?
It is never read and it might cause an overflow.

jhensel replied:
When I noticed that there may be an overflow either way, I reversed my changes and and moved this example into todo. So this is the original version.

dbeyer commented:
@peterschrammel Could you please review?

peterschrammel commented:
e38805e and f2ab0f7 could be given better commit messages or they could be squashed.

jhensel commented:
@peterschrammel I squashed all commits.
